editor/src/utils/utils.ts

```diff
@@ -837,36 +837,36 @@ function immutableUpdate(
   objPath: Array<string | number>,
   valueToSet: any,
 ): any {
-  switch (objPath.length) {
-    case 0:
-      // No path, so we're just replacing the whole value at this point.
-      return valueToSet
-    case 1:
-      // Last part of the path, setting the `valueToSet` where the final part specifies.
-      return immutableUpdateField(valueToUpdate, objPath[0], valueToSet)
-    default:
-      // 2 or more path elements, need to step down path part to recursively invoke this on the remainder.
-      const [first, ...remainder] = objPath
-      const fieldParsedAsNumber: number = typeof first === 'number' ? first : parseInt(first)
-      const isArrayUpdate = typeof first === 'number' || !isNaN(fieldParsedAsNumber)
-      if (isArrayUpdate) {
-        // Arrays.
-        const defaultedArray: Array<any> = defaultIfNull([], valueToUpdate)
-        let result: Array<any> = [...defaultedArray]
-        result[fieldParsedAsNumber] = immutableUpdate(
-          defaultedArray[fieldParsedAsNumber],
-          remainder,
-          valueToSet,
-        )
-        return result
-      } else {
-        // Objects.
-        const defaultedObject: { [key: string]: any } = defaultIfNull({}, valueToUpdate)
-        return {
-          ...defaultedObject,
-          [first]: immutableUpdate(defaultedObject[first], remainder, valueToSet),
-        }
+  const first = objPath[0]
```

Comment:
It looks like we're not using `immutableUpdate` anyway, so this whole function (as well as `immutableUpdateField`) can go in the bin